src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -112,28 +114,32 @@ public boolean validCheckForObject(final AtlasObject object)
         // will be flagged
         if (!complexRoundabout.isValid())
         {
-            // AutoFix candidate only for wrong direction case.
-            if (complexRoundabout.getAllInvalidations().size() == 1 && complexRoundabout
-                    .getAllInvalidations().get(0).getReason().equals(WRONG_WAY_INSTRUCTIONS))
+            // AutoFix candidate only for Single Way Roundabout with wrong direction.
+            if (complexRoundabout.getAllInvalidations().size() == 1
+                    && complexRoundabout.getAllInvalidations().get(0).getReason()
+                            .equals(WRONG_WAY_INSTRUCTIONS)
+                    && !this.isMultiWayRoundabout(roundaboutEdgeSet))
             {
                 // Mark that the Edges have been processed
                 roundaboutEdgeSet.forEach(
                         roundaboutEdge -> this.markAsFlagged(roundaboutEdge.getIdentifier()));
 
                 return Optional.of(this
-                        .createFlag(roundaboutEdgeSet,
+                        .createFlag(new OsmWayWalker((Edge) object).collectEdges(),
                                 this.getLocalizedInstruction(1, object.getOsmIdentifier()))
-                        .addFixSuggestion(FeatureChange.add(
-                                (AtlasEntity) ((CompleteEntity) CompleteEntity
-                                        .from((AtlasEntity) object)).withGeometry(
-                                                ((Edge) object).asPolyLine().reversed()),
-                                object.getAtlas())));
+                        .addFixSuggestion(
+                                FeatureChange.add(
+                                        (AtlasEntity) ((CompleteEntity) CompleteEntity
+                                                .from((AtlasEntity) object)).withGeometry(
+                                                        CommonMethods.buildOriginalOsmWayGeometry(
+                                                                (Edge) object).reversed()),
+                                        object.getAtlas())));
```

Comment:
This is an interesting way to combine the edges into a single fix suggestion. It could result in some nondeterminism due to taking on the atlas id of the starting object, but I think that is ok in this instance.